uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsenceType.php

```diff
@@ -219,15 +219,18 @@ private static function validateParams($params) {
    * method checks if one such type already exists and throws an error if that
    * is the case.
    *
+   * @param array $params
+   *
    * @throws \CRM_HRLeaveAndAbsences_Exception_InvalidAbsenceTypeException
-   * @throws \CiviCRM_API3_Exception
    */
-  private static function validateAddPublicHolidayToEntitlement() {
-    $result = civicrm_api3('AbsenceType', 'getcount', array(
-        'sequential' => 1,
-        'add_public_holiday_to_entitlement' => 1,
-    ));
-    if(!isset($result['result']) || $result['result'] > 0) {
+  private static function validateAddPublicHolidayToEntitlement($params) {
+    $dao = new self();
+    $dao->add_public_holiday_to_entitlement = 1;
+    if(!empty($params['id'])) {
+      $dao->whereAdd('id <> '. $params['id']);
```

Comment:
since It 09:30 PM here I may miss something , but I think you are passing  $params['id'] without escaping it.